rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -19,164 +19,332 @@
+
+            // save federations in storage
+            StorageAccessor storageAccessor = new InMemoryStorage();
+            storageProvider = new FederationStorageProviderImpl(storageAccessor);
+            storageProvider.setOldFederation(oldFederation);
+            storageProvider.setNewFederation(newFederation);
+        }
 
-        // Act
-        Federation activeFederation = federationSupport.getActiveFederation();
+        @ParameterizedTest
+        @MethodSource("expectedFederation_args")
+        void getActiveFederation_returnsExpectedFederationAccordingToActivationAgeAndActivations(long currentBlock, Federation expectedFederation,
+                                                                                                               ActivationConfig.ForBlock activations) {
+            Block executionBlock = mock(Block.class);
+            when(executionBlock.getNumber()).thenReturn(currentBlock);
+
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .withRskExecutionBlock(executionBlock)
+                .withActivations(activations)
+                .build();
+
+            Federation activeFederation = federationSupport.getActiveFederation();
+            assertThat(activeFederation, is(expectedFederation));
+        }
+        private Stream<Arguments> expectedFederation_args() {
+            return Stream.of(
+                Arguments.of(blockNumberFederationActivationHop - 1, oldFederation, hopActivations), // new federation shouldn't be active
+                Arguments.of(blockNumberFederationActivationHop + 1, newFederation, hopActivations), // new federation should be active
+                Arguments.of(blockNumberFederationActivationFingerroot - 1, oldFederation, fingerrootActivations), // new federation shouldn't be active
+                Arguments.of(blockNumberFederationActivationFingerroot + 1, newFederation, fingerrootActivations) // new federation should be active
+            );
+        }
 
-        // Assert
-        if (newFedExpectedToBeActive){
-            assertThat(activeFederation, is(newFederation));
-        } else {
-            assertThat(activeFederation, is(oldFederation));
+        @Test
+        void getActiveFederationRedeemScript_beforeRSKIP293_returnsEmpty() {
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .build();
+            assertFalse(federationSupport.getActiveFederationRedeemScript().isPresent());
+        }
+
+        @ParameterizedTest
+        @MethodSource("expectedRedeemScript_args")
+        void getActiveFederationRedeemScript_returnsExpectedRedeemScriptAccordingToActivationAgeAndActivations(long currentBlock, Script expectedRedeemScript,
```

Comment:
Here and in other tests, I'd put the expected as the last param. Makes it easier to understand for me at least. First the setup values then finally the expected